src/cascadia/TerminalCore/ICoreAppearance.idl

```diff
@@ -50,6 +50,37 @@ namespace Microsoft.Terminal.Core
         UInt32 Value;
     };
 
+    struct Scheme
+    {
+        Microsoft.Terminal.Core.Color Foreground;
+        Microsoft.Terminal.Core.Color Background;
+
+        Microsoft.Terminal.Core.Color SelectionBackground;
+
+        Microsoft.Terminal.Core.Color CursorColor;
+
+        // Table: A WinRT struct doesn't allow pointers (like an array) in
+        // structs, but we very much would like this object to be a struct. So
+        // we'll call out each color individually. There's only 16, it's not
+        // that bad.

```

Comment:
These colors are the well known, standard 16 colors, right? If you make this a simple array `Color[16]` you can safe yourself a lot of redundant code elsewhere.

Reply:
Unfortunately, MIDL3 doesn't allow arrays in a struct. That's what i was trying to go for in the above comment. If you do this the sensible way, you'll get 
```
error MIDL4000: [msg]A structure field cannot be a type of pointer. [context]: Colors [ Field 'Colors' of Struct 'Microsoft.Terminal.Core.Scheme'  ]
```

Reply:
Honestly if I were you I'd put those 16 members into a separate struct like `BaseColorTable` and then simply write a serialization "swizzler" to cast that struct to an array like so:
```
inline std::array<Core::Color, 16>& _swizzleBaseColortable(BaseColorTable& table) noexcept
{
    static_assert(sizeof(table) == sizeof(std::array<Core::Color, 16>));
    static_assert(alignof(table) == alignof(std::array<Core::Color, 16>));
    return *reinterpret_cast<std::array<Core::Color, 16>*>(&table);
}
```

Afterwards you can write loops and stuff and get rid of some of the more repetitive code.